File: result/translate/transcoder/python2java/java/MAKE_LARGEST_PALINDROME_CHANGING_K_DIGITS.java
public static String f_gold ( String str , int k ) {
  char [ ] palin = Arrays . copyOf ( str . toCharArray ( ) , str . length ( ) ) ;
  int l = 0 ;
  int r = str . length ( ) - 1 ;
  while ( ( l <= r ) && ( ( l < str . length ( ) ) || ( r < str . length ( ) ) ) ) {
    if ( ( str . charAt ( l ) != str . charAt ( r ) ) && ( str . charAt ( l ) != str . charAt ( r ) ) ) {
      palin [ l ] = palin [ r ] = Math . max ( str . charAt ( l ) , str . charAt ( r ) ) ;
      k -- ;
    }
    l ++ ;
    r -- ;
  }
  if ( ( k < 0 ) || ( k > 0 ) ) return "Not possible" ;
  l = 0 ;
  r = str . length ( ) - 1 ;
  while ( ( l <= r ) && ( ( l < str . length ( ) ) || ( r < str . length ( ) ) ) ) {
    if ( ( l == r ) && ( ( k > 0 ) || ( k < 0 ) ) ) palin [ l ] = palin [ r ] = '9' ;
    if ( ( palin [ l ] < '9' ) || ( palin [ l ] == str . charAt ( l ) && ( palin [ r ] == str . charAt ( r ) ) ) ) ) {
      k -- ;
      palin [ l ] = palin [ r ] = '9' ;
    }
    else if ( ( k >= 2 ) && ( palin [ l ] != str . charAt ( l ) || palin [ r ] != str . charAt ( r ) ) ) ) {
      k -- ;
      palin [ l ] = palin [ r ] = '9' ;
    }
  }
  l ++ ;
  r -- ;
}
return "" ;
}

File: result/translate/chatgpt/java2cpp/c++/MAKE_LARGEST_PALINDROME_CHANGING_K_DIGITS.cpp
std::string f_gold(std::string str, int k) {
    std::string ans = "";
    std::vector<char> palin(str.begin(), str.end());
    int l = 0;
    int r = str.length() - 1;
    while (l < r) {
        if (str[l] != str[r]) {
            palin[l] = palin[r] = std::max(str[l], str[r]);
            k--;
        }
        l++;
        r--;
    }
    if (k < 0) {
        return "Not possible";
    }
    l = 0;
    r = str.length() - 1;
    while (l <= r) {
        if (l == r) {
            if (k > 0) {
                palin[l] = '9';
            }
        }
        if (palin[l] < '9') {
            if (k >= 2 && palin[l] == str[l] && palin[r] == str[r]) {
                k -= 2;
                palin[l] = palin[r] = '9';
            } else if (k >= 1 && (palin[l] != str[l] || palin[r] != str[r])) {
                k--;
                palin[l] = palin[r] = '9';
            }
        }
        l++;
        r--;
    }
    for (int i = 0; i < palin.size(); i++) {
        ans += palin[i];
    }
    return ans;
}